waffle_hub/utils/memory.py

```diff
@@ -0,0 +1,18 @@
+import torch
+
+
+def use_cuda(device):
+    """For gpu memory decorator"""
+
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            if torch.cuda.is_available():
+                torch.cuda.init()  # for thread
+            func(*args, **kwargs)
+            if torch.cuda.is_available() and device != "cpu":
+                # Memory free
+                torch.cuda.empty_cache()
```

Comment:
Is `and device != "cpu"` necessary?
And the naming of `use_cuda` function looks awkward.
How aboud `init_device` or `device_context` or something?
Becuase, there are many chances to add some device settings in the future (like cpu or other hardware dependancies)

Reply:
I included `device=='cpu'` to prevent GPU from trying to use it as `CPU` on available devices.
Because if other threads are using "GPU", we shouldn't try `empty_cache()` when using "CPU".

Looking at this comment, it seems that the name "Use_cuda" was not appropriate.
Therefore, I changed the name to "device_context".